src/services/Libs.ts

```diff
@@ -151,6 +151,134 @@ export const extractMainDomain = (domain: string): string => {
   return `${partsOfDomain.slice(length - 2).join('.')}${eDot ? '.' : ''}`;
 };
 
+/**
+ * This fetches all (first party) cookies for a given tab domain
+ * @param state The webextension state
+ * @param tab The tab to fetch all (first party) cookies for.
+ */
+export const getAllCookiesForDomain = async (
+  state: State,
+  tab: browser.tabs.Tab,
+): Promise<browser.cookies.Cookie[] | undefined> => {
+  if (!tab.url || tab.url === '') return;
+  if (tab.url.startsWith('about:') || tab.url.startsWith('chrome:')) return;
+  const debug = getSetting(state, 'debugMode') as boolean;
+  const partialTabInfo = createPartialTabInfo(tab);
+  const { cookieStoreId, url } = tab;
+  const hostname = getHostname(url);
+  if (hostname === '') {
+    cadLog(
+      {
+        msg: 'Libs.getAllCookiesForDomain:  hostname parsed empty for tab url.',
+        x: { partialTabInfo, hostname },
+      },
+      debug,
+    );
+    return;
+  }
+  let cookies: browser.cookies.Cookie[] = [];
+  const mainDomain = extractMainDomain(hostname);
+
+  if (hostname.startsWith('file:')) {
+    const allCookies = await browser.cookies.getAll(
+      returnOptionalCookieAPIAttributes(state, {
+        storeId: cookieStoreId,
+      }),
+    );
+    const regExp = new RegExp(hostname.slice(7)); // take out 'file://'
+    cadLog(
+      {
+        msg:
+          'Libs.getAllCookiesForDomain:  Local File Regex to rest on cookie.path',
+        x: { partialTabInfo, hostname, regExp: regExp.toString() },
+      },
+      debug,
+    );
+    cookies = allCookies.filter((c) => c.domain === '' && regExp.test(c.path));
+  } else {
+    cadLog(
+      {
+        msg: 'Libs.getAllCookiesForDomain:  browser.cookies.getAll for domain.',
+        x: {
+          partialTabInfo,
+          domain: hostname,
+          firstPartyDomain: mainDomain,
+        },
+      },
+      debug,
+    );
+    cookies = await browser.cookies.getAll(
+      returnOptionalCookieAPIAttributes(state, {
+        domain: hostname,
+        firstPartyDomain: mainDomain,
+        storeId: cookieStoreId,
+      }),
+    );
+  }
+  cadLog(
+    {
+      msg: 'Libs.getAllCookiesForDomain:  Filtered Cookie Count',
+      x: {
+        partialTabInfo,
+        tabURL: tab.url,
+        hostname,
+        cookieCount: cookies.length,
+      },
+    },
+    debug,
+  );
+
+  if (!isFirefox(state.cache)) return cookies;
+  // Firefox only - try to get additional firstParty Isolation cookies
+  // if firstparty.isolation.use_site was enabled, to which we don't know
+  const siteURL = new URL(url);
+  const proto = siteURL.protocol.replace(':', '');
+  // firstPartyDomain = (https,domain.com)
+  cadLog(
+    {
+      msg: 'Libs.getAllCookiesForDomain:  browser.cookies.getAll for domain.',
+      x: {
+        partialTabInfo,
+        domain: hostname,
+        firstPartyDomain: `(${proto},${mainDomain})`,
+      },
+    },
+    debug,
+  );
+  const cookies_fpi_usesite = await browser.cookies.getAll(
```

Comment:
camelCase for consistency? like `cookiesFPIUsesSite` or something

Reply:
Resolved in 
097c8cf